delivery_product_restriction/models/sale_order.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2022 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    @api.depends("partner_shipping_id", "order_line")
```

Comment:
Why no `@api.multi ?`

Reply:
I think that `@api.multi` is the default. Many `_compute` method, only specified `@api.depends` an not `@api.multi, but should work with a recorset with several elements.

Reply:
`multi` is not the default, it prepares the self with the old style `cr, uid, ids` tuples. If the self/recordset has already been called through a `multi`, the self is already prepared.

As you wrote, many `depends` don't specify `multi` so I guess the orm calls the functions decorated by `depends` on initialized recordsets.

So good here 👍

Reply:
Thanks for this precision. :) I learn every day ! :)